usaspending_api/search/elasticsearch/filters/tas.py

```diff
@@ -0,0 +1,110 @@
+
+    @staticmethod
+    def string_to_dictionary(string):
+        if len(string.split("-")) == 1:
+            return {"agency": string}
+        elif len(string.split("-")) == 2:
+            return FederalAccount.fa_rendering_label_to_component_dictionary(string)
+        else:
+            return TreasuryAppropriationAccount.tas_rendering_label_to_component_dictionary(string)
+
+
+def search_regex_of(v):
+    if isinstance(v, str):
+        v = TasCodes.string_to_dictionary(v)
+
+    code_lookup = {
+        "agency": f"agency={v['agency']}" if v.get("agency") else "*",
+        "faaid": f"faaid={v['faaid']}" if v.get("faaid") else "*",
+        "famain": f"famain={v['famain']}" if v.get("famain") else "*",
+        "aid": f"aid={v['aid']}" if v.get("aid") else "*",
+        "main": f"main={v['main']}" if v.get("main") else "*",
+        "ata": f"ata={v['ata']}" if v.get("ata") else "*",
+        "sub": f"sub={v['sub']}" if v.get("sub") else "*",
+        "bpoa": f"bpoa={v['bpoa']}" if v.get("bpoa") else "*",
+        "epoa": f"epoa={v['epoa']}" if v.get("epoa") else "*",
+        "a": f"a={v['a']}" if v.get("a") else "*",
+    }
+
+    # This is NOT the order of elements as displayed in the tas rendering label, but instead the order in the award_delta_view and transaction_delta_view
+    search_regex = (
+        code_lookup["agency"]
+        + code_lookup["faaid"]
+        + code_lookup["famain"]
+        + code_lookup["aid"]
+        + code_lookup["main"]
+        + code_lookup["ata"]
+        + code_lookup["sub"]
+        + code_lookup["bpoa"]
+        + code_lookup["epoa"]
+        + code_lookup["a"]
+    )
+
+    # TODO: move this to a Tinyshield filter
+    if not re.match(r"^(\d|\w|-|\*|=)+$", search_regex):
+        raise UnprocessableEntityException(f"Unable to parse TAS filter")
+
+    print(search_regex)
+    return search_regex
+
+
+class TASNode(Node):
+    def _basic_search_unit(self):
+        return search_regex_of(self.code)
+
+    def clone(self, code, positive, positive_naics, negative_naics):
+        return TASNode(code, positive, positive_naics, negative_naics)
+
+
+class TreasuryAccounts(_Filter):
+    underscore_name = "treasury_account_components"
+
+    @classmethod
+    def generate_elasticsearch_query(cls, filter_values: list, query_type: _QueryType) -> ES_Q:
+        tas_codes_query = []
+
+        for v in filter_values:
+            code_lookup = {
+                "aid": v.get("aid", ".*"),
+                "main": v.get("main", ".*"),
+                "ata": v.get("ata", ".*"),
+                "sub": v.get("sub", ".*"),
+                "bpoa": v.get("bpoa", ".*"),
+                "epoa": v.get("epoa", ".*"),
+                "a": v.get("a", ".*"),
+            }
+
+            search_regex = f".*aid={code_lookup['aid']}main={code_lookup['main']}ata={code_lookup['ata']}sub={code_lookup['sub']}bpoa={code_lookup['bpoa']}epoa={code_lookup['epoa']}a={code_lookup['a']}"
```

Comment:
I'm curious about performance after adding a leading wildcard to the regex. @sethstoudenmier @emily-brents is this OK from a performance perspective?

Reply:
Generally no, we want to avoid leading wildcards specifically because they cause performance hits - from the Elasticsearch docs: 

> Avoid beginning patterns with * or ?. This can increase the iterations needed to find matching terms and slow search performance.

Reply:
Would it be alright to just add another column to the index? There is no way to order the tas codes to never start with a wildcard, but we could always keep adding different strings to regex on.

Reply:
If I understand the searches, FA and TAS should be two separate fields in the index.

Reply:
So there are two searches: the component search (which this line of code is doing), and the search by path (in the class above in this file). In the component search, Federal Account does not matter; only TAS components are searched on. In the search by path, the agency, federal account, and TAS all matter. Right now, this index column `treasury_accounts` holds data for all three, in order  to support both searches with one index. Right now I'm experimenting locally with a second index with just TAS data to support the component search without needing to start with a wildcard.

Reply:
Keep in mind that Federal Account is a property of a TAS, not an Award or a Transaction.

Reply:
> Keep in mind that Federal Account is a property of a TAS, not an Award or a Transaction.

But the data are pulled from different tables/columns because they don't match perfectly, right? Otherwise the new "FA" fields wouldn't be needed since they would be duplicating pre-existing TAS fields.

Reply:
Yes, they are a join from the tas table to fa and agency. What I was trying to say is that they aren't one-to-one with awards or transactions, so even if it was a new column in the index, it would still need to be associated with tas.